internal/storage/storage_handle.go

```diff
@@ -142,9 +144,17 @@ func createClientOptionForGRPCClient(ctx context.Context, clientConfig *storageu
 
 	clientOpts = append(clientOpts, option.WithGRPCConnectionPool(clientConfig.GrpcConnPoolSize))
 	clientOpts = append(clientOpts, option.WithUserAgent(clientConfig.UserAgent))
-	// Turning off the go-sdk metrics exporter to prevent any problems.
-	// TODO (kislaykishore) - to revisit here for monitoring support.
-	clientOpts = append(clientOpts, storage.WithDisabledClientMetrics())
+
+	// Pass the OpenTelemetry MeterProvider to the Go storage client,
+	// using the new WithMeterProvider client option.
+	// TODO - detect when running on GKE with clientConfig.IsGKE.
+	if clientConfig.EnableGrpcMetrics {
```

Comment:
If this is enable-grpc-metrics is false, let's continue the old behavior - `clientOpts = append(clientOpts, storage.WithDisabledClientMetrics())`. If enable-grpc-metrics is true, and isGKE is false, then skip setting `clientOpts = append(clientOpts, storage.WithDisabledClientMetrics())`